corehq/apps/userreports/reports/builder/forms.py

```diff
@@ -425,14 +425,16 @@ def indicators(self, columns, filters, is_multiselect_chart_report=False, as_dic
                     # will use one indicator for the field's string value, and "Sum" and "Average" aggregations
                     # will use a second indicator for the field's numerical value. "column_id" includes the
                     # indicator's data type, so it is unique per indicator.
-                    indicators.setdefault(indicator['column_id'], indicator)
+                    indicator_key = (indicator['column_id'], indicator['type'])
```

Comment:
@czue the failing test checked that indicators for a multi-select question included a "choice_list" indicator. The test failed because I assumed that the column_id for indicators of the same data source would be unique, but that assumption was wrong. 

These are the indicators: 
```javascript
{
  "column_id": "data_state_6e36b993", 
  "datatype": "string", 
  "display_name": "state", 
  "expression": {
    "property_path": [
      "form", 
      "state"
    ], 
    "type": "property_path"
  }, 
  "type": "expression"
}
{
  "choices": [
    "MA", 
    "MN", 
    "VT"
  ], 
  "column_id": "data_state_6e36b993", 
  "display_name": "state", 
  "property_path": [
    "form", 
    "state"
  ], 
  "select_style": "multiple", 
  "type": "choice_list"
}
```

So when this keyed the `indicators` dictionary on `indicator['column_id']`, the choice_list indicator was being replaced with the the expression indicator.

I'm worried that my assumption should be true though. Is this a bug? Should column_id be unique?

Reply:
the current validation checks that [database columns should be unique](https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/userreports/models.py#L330). since choice list indicators get expanded to [one column per choice](https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/userreports/indicators/factory.py#L102) it's not actually creating a conflict.

this is definitely confusing and i'd support if someone wanted to make the case that `column_id` should be unique within a data source even if it's not strictly required, but perhaps might be a bit of a detour - one potentially requiring complex data migrations - and not critical for the current work?

Reply:
Thanks. For now, at least, I'm happy with just knowing why the current behaviour is fine.